Unity-Technologies/Graphics
# Hd/add layer modification on generated emissive mesh for area light

### Purpose of this PR
Add missing field to change the Layer on generated Emissive Mesh

![image](https://user-images.githubusercontent.com/40034005/81176963-397c1e00-8fa6-11ea-9f3c-9ddd4683afbb.png)


- Added hability to have same layer than Light one and use it by default
- Added fence to prevent moving the non editable generated EmissiveMesh in the hierarchy while in edit mode
- Added auto sync of "static" for generated mesh GameObject to be the same than the Light GameObject (and also the static drop down)

---
### Testing status

**Manual Tests**: 
- Other: Manually tested that:
  - Layer of generated Emissive Mesh change according to the field change
  - creation of a generated Emissive Mesh use the correct Layer
  - when changing Layer of an existing EmissiveMesh by changing one parent GameObject and saying to apply the change to all childs, the change is discarded. Note that it is discarded at generation and the generated element is never saved/serialized. In Editor, if changing it by changing a parent, and then going directly to the child Emissive Mesh, it **can display the wrong value**. This value is reset first time you select the light again (or if the light was the parent that caused the change, first time the Inspector is redraw, which happens each time you change one thing in it)

**Not Performed:** Test while manipulating a prefab of the Light. Though, I do not introduce new logic and rely on the same than what have worked for the other parameter for the Emissive Mesh

**Automated Tests**: 

---
### Comments to reviewers
**[do not merge]** wainting for test (and @peeweek answear) to be sure everything is catched
